spring-integration-sftp/src/main/java/org/springframework/integration/sftp/session/DefaultSftpSessionFactory.java

```diff
@@ -191,8 +193,21 @@ public void setPassword(String password) {
 	 * false (default).</b>
 	 * @param knownHosts The known hosts.
 	 * @see JSch#setKnownHosts(String)
+	 * @deprecated since 5.2.5 in favor of {@link #setKnownHosts(Resource)}
 	 */
+	@Deprecated
 	public void setKnownHosts(String knownHosts) {
+		setKnownHosts(new FileSystemResource(knownHosts));
+	}
+
+	/**
+	 * Specifies the filename that will be used for a host key repository.
+	 * The file has the same format as OpenSSH's known_hosts file.
+	 * @param knownHosts the resource for known hosts.
+	 * @see JSch#setKnownHosts(InputStream)
+	 * @since 5.2.5
+	 */
+	public void setKnownHosts(Resource knownHosts) {
```

Comment:
Won't this cause a problem with `<bean/>` config? (ambiguous setter) `setKnownHostsResource()` ?

Reply:
Looks like Juergen has fixed something in SF. See Travis 😄

Reply:
If I recall correctly, failures were random; also it looks like you changed all the tests to return a `Resource` using SpEL.

Perhaps the problem only occurs when a PropertyEditor is needed (although I remember most of the problems were with String Vs. Expression .

I will defer to you, though, if you think it's safe.

Reply:
Well, I can rename it to `setKnownHostsResource` to have it really very safe...